--- calendar/event.php	Fri Jan 30 12:35:40 2015
+++ calendar/event.php	Wed Feb 04 10:33:23 2015
@@ -42,12 +42,14 @@
  * This file is part of the Calendar section Moodle
  *
  * @copyright 2003-2004 Jon Papaioannou (pj@moodle.org)
+ * @copyright 2015 Simon Bosman (simonbosman@gmail.com)
  * @license http://www.gnu.org/copyleft/gpl.html GNU GPL v2 or later
  * @package calendar
  */
 
 require_once('../config.php');
 require_once($CFG->dirroot.'/calendar/event_form.php');
+require_once($CFG->dirroot.'/calendar/event_form_homework.php');
 require_once($CFG->dirroot.'/calendar/lib.php');
 require_once($CFG->dirroot.'/course/lib.php');
 
@@ -57,6 +59,8 @@
 $eventid = optional_param('id', 0, PARAM_INT);
 $courseid = optional_param('courseid', SITEID, PARAM_INT);
 $courseid = optional_param('course', $courseid, PARAM_INT);
+$groupid = optional_param('groupid', $groupid, PARAM_INT);
+$homework = optional_param('homework', False, PARAM_BOOL);
 $day = optional_param('cal_d', 0, PARAM_INT);
 $month = optional_param('cal_m', 0, PARAM_INT);
 $year = optional_param('cal_y', 0, PARAM_INT);
@@ -127,31 +131,40 @@
     }
 } else {
     $title = get_string('newevent', 'calendar');
-    calendar_get_allowed_types($formoptions->eventtypes, $course);
     $event = new stdClass();
+    if ($homework){
+    	global $DB;
+    	$group = $DB->get_record('groups', array('id'=>$groupid));
+    	$event->eventtype = 'course';
+        $title = 'Huiswerk opgeven';
+    }
+    calendar_get_allowed_types($formoptions->eventtypes, $course);
     $event->action = $action;
     $event->course = $courseid;
     $event->courseid = $courseid;
+    $event->groupid = $groupid;
+    $event->groupname = $group->name;
     $event->timeduration = 0;
     if ($formoptions->eventtypes->courses) {
-        if (!$issite) {
+     	if (!$issite) {
             $event->eventtype = 'course';
-        } else {
-            unset($formoptions->eventtypes->courses);
+        } elseif (!homework) {
+         	unset($formoptions->eventtypes->courses);
             unset($formoptions->eventtypes->groups);
         }
     }
-    $event->timestart = $time;
+  	$event->timestart = $time;
     $event = new calendar_event($event);
     if (!calendar_add_event_allowed($event)) {
         print_error('nopermissions');
     }
 }
-
 $properties = $event->properties(true);
 $formoptions->event = $event;
 $formoptions->hasduration = ($event->timeduration > 0);
-$mform = new event_form(null, $formoptions);
+if($homework)
+	$properties->eventtype='group';
+$mform =($homework)? new event_form_homework(null, $formoptions) : new event_form(null, $formoptions);
 $mform->set_data($properties);
 $data = $mform->get_data();
 if ($data) {
--- login/index.php	Fri Jan 30 12:35:40 2015
+++ login/index.php	Fri Jan 30 15:42:06 2015
@@ -84,6 +84,8 @@
 $user = false;
 
 $authsequence = get_enabled_auth_plugins(true); // auths, in sequence
+if ($authsequence != null)
+	 $authsequence = array_reverse($authsequence);
 foreach($authsequence as $authname) {
     $authplugin = get_auth_plugin($authname);
     $authplugin->loginpage_hook();
